src/ripple/ledger/impl/ApplyStateTable.cpp

```diff
@@ -520,12 +522,24 @@ ApplyStateTable::destroyXRP(XRPAmount const& fee)
 
 // Insert this transaction to the SLE's threading list
 void
-ApplyStateTable::threadItem(TxMeta& meta, std::shared_ptr<SLE> const& sle)
+ApplyStateTable::threadItem(
+    TxMeta& meta,
+    std::shared_ptr<SLE> const& sle,
+    ReadView const& view)
 {
     key_type prevTxID;
     LedgerIndex prevLgrID;
 
-    if (!sle->thread(meta.getTxID(), meta.getLgrSeq(), prevTxID, prevLgrID))
+    static std::set<LedgerEntryType> newPreviousTxnIDTypes = {
+        ltDIR_NODE, ltAMENDMENTS, ltFEE_SETTINGS, ltNEGATIVE_UNL, ltAMM};
+    bool const includePrevTxnID = view.rules().enabled(fixPreviousTxnID) ||
+        !newPreviousTxnIDTypes.count(sle->getType());
```

Comment:
I was going to suggest rewriting this so that it would be clearer that everything that had the fields was fine to thread except these new types
```
bool const excludePrevTxnID = !view.rules().enabled(fixPreviousTxnID) &&
    newPreviousTxnIDTypes.count(sle->getType());
```
Then I realized that this was exactly the boolean NOT of what you have.

Nothing I'm asking you to change, I just wanted to make the observation.

Reply:
Fixed (and clarified with a comment) in f5a7cc4b5983019ff8763ba68679a20f96b36212

Reply:
Yeah, that ended up improving readability way more than I expected.